src/test/java/org/kie/processmigration/service/SchedulerServiceImplTest.java

```diff
@@ -0,0 +1,212 @@
+
+    @Inject
+    SchedulerService schedulerService;
+
+    @InjectMock
+    MigrationService migrationService;
+
+    @Inject
+    Scheduler scheduler;
+
+    @AfterEach
+    void resetScheduler() throws SchedulerException {
+        if(scheduler != null) {
+            scheduler.clear();
+        }
+    }
+
+    @Test
+    void testScheduler() {
+        assertThat(schedulerService, notNullValue());
+    }
+
+    @Test
+    void testScheduleMigrationNow() throws InvalidMigrationException, InterruptedException, MigrationNotFoundException, SchedulerException {
+        // Given
+        MigrationDefinition definition = new MigrationDefinition();
+        definition.setPlanId(1L);
+        definition.setKieServerId("kie-server-1");
+        definition.setRequester("me");
+        definition.setProcessInstanceIds(List.of(1L));
+        Migration migration = new Migration(definition);
+        migration.id = 99L;
+        CountDownLatch count = new CountDownLatch(1);
+        when(migrationService.get(migration.id)).thenReturn(migration);
+        scheduler.getListenerManager().addJobListener(new TestJobListener(count), allJobs());
+
+        // When
+        schedulerService.scheduleMigration(migration);
+
+        // Then
+        count.await(10, TimeUnit.SECONDS);
+        verify(migrationService, times(1)).get(99L);
+        verify(migrationService, times(1)).migrate(migration);
+        assertThat(scheduler.checkExists(new JobKey(migration.id.toString())), is(Boolean.FALSE));
+        assertThat(scheduler.checkExists(new TriggerKey(migration.id.toString())), is(Boolean.FALSE));
+    }
+
+    @Test
+    void testScheduleMigration() throws InvalidMigrationException, InterruptedException, MigrationNotFoundException, SchedulerException {
+        // Given
+        MigrationDefinition definition = new MigrationDefinition();
+        definition.setPlanId(1L);
+        definition.setKieServerId("kie-server-1");
+        definition.setRequester("me");
+        definition.setProcessInstanceIds(List.of(1L));
+        Instant when = Instant.now().plus(500, ChronoUnit.MILLIS);
+        definition.setExecution(new Execution().setType(Execution.ExecutionType.ASYNC).setScheduledStartTime(when));
+        Migration migration = new Migration(definition);
+        migration.id = 99L;
+        CountDownLatch count = new CountDownLatch(1);
+        scheduler.getListenerManager().addJobListener(new TestJobListener(count), allJobs());
+        when(migrationService.get(migration.id)).thenReturn(migration);
+
+        // When
+        schedulerService.scheduleMigration(migration);
+
+        // Then
+        assertThat(scheduler.checkExists(new JobKey(migration.id.toString())), is(Boolean.TRUE));
+        assertThat(scheduler.checkExists(new TriggerKey(migration.id.toString())), is(Boolean.TRUE));
+        count.await(10, TimeUnit.SECONDS);
+
+        verify(migrationService, times(1)).get(99L);
+        verify(migrationService, times(1)).migrate(migration);
+    }
```

Comment:
why not using the same logic as above before test method to check the job no longer exists?

```suggestion
        assertThat(scheduler.checkExists(new JobKey(migration.id.toString())), is(Boolean.FALSE));
        assertThat(scheduler.checkExists(new TriggerKey(migration.id.toString())), is(Boolean.FALSE));
    }
```